src/libData/AccountData/AccountStoreSC.tpp

```diff
@@ -1149,200 +1148,200 @@ bool AccountStoreSC<MAP>::ParseCallContractJsonOutput(
-      receipt.AddError(RECEIPT_IS_NULL);
-      return false;
-    }
+      try {
+        m_curAmount = boost::lexical_cast<uint128_t>(msg["_amount"].asString());
+      } catch (...) {
+        LOG_GENERAL(WARNING, "_amount " << msg["_amount"].asString()
+                                        << " is not numeric");
+        return false;
+      }
 
-    account = m_accountStoreAtomic->GetAccount(recipient);
+      recipient = Address(msg["_recipient"].asString());
+      if (IsNullAddress(recipient)) {
+        LOG_GENERAL(WARNING, "The recipient can't be null address");
+        receipt.AddError(RECEIPT_IS_NULL);
+        return false;
+      }
 
-    if (account == nullptr) {
-      AccountStoreBase<MAP>::AddAccount(recipient, {0, 0});
       account = m_accountStoreAtomic->GetAccount(recipient);
-    }
 
-    // Recipient is non-contract
-    if (!account->isContract()) {
-      LOG_GENERAL(INFO, "The recipient is non-contract");
-      if (!TransferBalanceAtomic(m_curContractAddr, recipient, m_curAmount)) {
-        receipt.AddError(BALANCE_TRANSFER_FAILED);
-        return false;
-      } else {
+      if (account == nullptr) {
+        AccountStoreBase<MAP>::AddAccount(recipient, {0, 0});
+        account = m_accountStoreAtomic->GetAccount(recipient);
+      }
+
+      // Recipient is non-contract
+      if (!account->isContract()) {
+        LOG_GENERAL(INFO, "The recipient is non-contract");
+        if (!TransferBalanceAtomic(m_curContractAddr, recipient, m_curAmount)) {
+          receipt.AddError(BALANCE_TRANSFER_FAILED);
+          return false;
+        } else {
+          ret = true;
+        }
+      }
+
+      // Recipient is contract
+      // _tag field is empty
+      if (msg["_tag"].asString().empty()) {
+        LOG_GENERAL(INFO,
+                    "_tag in the scilla output is empty when invoking a "
+                    "contract, transaction finished");
         ret = true;
       }
-    }
 
-    // Recipient is contract
-    // _tag field is empty
-    if (_json["message"]["_tag"].asString().empty()) {
-      LOG_GENERAL(INFO,
-                  "_tag in the scilla output is empty when invoking a "
-                  "contract, transaction finished");
-      ret = true;
-    }
-  }
+      m_storageRootUpdateBufferAtomic.emplace(m_curContractAddr);
 
-  m_storageRootUpdateBufferAtomic.emplace(m_curContractAddr);
+      if (ENABLE_CHECK_PERFORMANCE_LOG) {
+        LOG_GENERAL(DEBUG,
+                    "LDB Write (microseconds) = " << r_timer_end(tpStart));
+        LOG_GENERAL(DEBUG, "Gas used = " << (startGas - gasRemained));
+      }
 
-  if (ENABLE_CHECK_PERFORMANCE_LOG) {
-    LOG_GENERAL(DEBUG, "LDB Write (microseconds) = " << r_timer_end(tpStart));
-    LOG_GENERAL(DEBUG, "Gas used = " << (startGas - gasRemained));
-  }
+      if (ret) {
+        // return true;
+        continue;
+      }
 
-  if (ret) {
-    return true;
-  }
+      LOG_GENERAL(INFO, "Call another contract");
+      receipt.AddDepth();
```

Comment:
Is this cleared when a new member in the `message` is accessed??

Reply:
the depth now actually means how many times of interpreter being called.

Reply:
Would not that lead to issues in error reporting ? anyways not a blocker

Reply:
it will just tell user your error is happened in which transition, basically this line is not newly added, I just moved it from another line to here